### PR TITLE
samples: nrf9160: Add a nRF9160 NIDD sample

### PR DESCRIPTION
**Pull request by stig-bjorlykke**
Add a nRF9160 sample using Non-IP Data Delivery (NIDD).

Signed-off-by: Stig Bjørlykke <stig.bjorlykke@nordicsemi.no>

### CODE REVIEW

File: samples/nrf9160/nidd/README.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+.. include:: /includes/tfm.txt
+
+Overview
+********
+
+The NIDD sample creates a Non-IP PDN using the configured APN, and uses socket operations to send and receive data.
```
Comment (umapraseeda) [inline samples/nrf9160/nidd/README.rst:24]:
```suggestion
The NIDD sample creates a Non-IP PDN using the configured APN and uses socket operations to send and receive data.
```

File: samples/nrf9160/nidd/README.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+   RRC mode: Connected
+   Network registration status: Connected - roaming
+   Get PDN ID 0
+   Created socket 0
+   Sent 13 bytes
+   Received 14 bytes: Hello, Device!
+   Closed socket 0
+   LTE cell changed: Cell ID: -1, Tracking area: -1
+   RRC mode: Idle
+   NIDD sample done
```
Comment (umapraseeda) [inline samples/nrf9160/nidd/README.rst:90]:
```suggestion
#. Observe that the sample starts and shows the following output from the device.
   Note that this is an example, and the output need not be identical to your observed output.

   .. code-block:: console

      NIDD sample started
      Configured Non-IP for APN "iot.nonip"
      LTE cell changed: Cell ID: 21657858, Tracking area: 40401
      RRC mode: Connected
      Network registration status: Connected - roaming
      Get PDN ID 0
      Created socket 0
      Sent 13 bytes
      Received 14 bytes: Hello, Device!
      Closed socket 0
      LTE cell changed: Cell ID: -1, Tracking area: -1
      RRC mode: Idle
      NIDD sample done
```

**Comment by umapraseeda** [comment]
LGTM! Shouldn't we add a changelog entry here: https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-10163/nrf/releases/release-notes-changelog.html#nrf9160-samples? May be something like this:
\* Added:

   \* :ref:\`nidd_sample\` sample that demonstrates how to use Non-IP Data Delivery (NIDD) on an nRF9160 device.

File: samples/nrf9160/nidd/README.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+.. _CONFIG_NIDD_ALLOC_NEW_CID:
+
+CONFIG_NIDD_ALLOC_NEW_CID - Allocate new context identifier for NIDD connection
+   This option, when enabled, allocates a new PDN context identifier instead of modifying the default.
+   This enables the use of NIDD together with regular IP traffic.
+
+.. CONFIG_NIDD_PAYLOAD:
+
+CONFIG_NIDD_PAYLOAD - Payload for NIDD transmission
+   This options sets the application payload to send as data.
```
Comment (umapraseeda) [inline samples/nrf9160/nidd/README.rst:57]:
```suggestion
   This option sets the application payload to send as data.
```

**Review by umapraseeda — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Pull request merged**